1. Tabs changed to Fragments

LizavetaVanahel opened the pull request.

LizavetaVanahel committed:
1. Tabs chenged to Fragments

PavelKapustsin commented on app/src/main/java/com/example/vanahel/tasksmanagerapplication/adapter/ViewPagerAdapter.java:
```diff
@@ -0,0 +1,39 @@
+
+    @Override
+    public int getCount() {
+        return tabCount;
```
getCount() method should be managed by ViewPagerAdapter itself, not by a param from constructor. Because if a programer sets incorrect value into constructor (by mistake) the app will crash, because a ViewPager will try to instantiate a gragment with index 2 and getItem() method will return null in this situation.

PavelKapustsin replied:
getCount() method should be managed by ViewPagerAdapter itself, not by a param from constructor. Because if a programer sets incorrect value into constructor (by mistake) the app will crash, because a ViewPager will try to instantiate a gragment with index 2 and getItem() method will return null in this situation.

PavelKapustsin commented on app/src/main/java/com/example/vanahel/tasksmanagerapplication/adapter/ViewPagerAdapter.java:
```diff
@@ -0,0 +1,39 @@
+    public Fragment getItem(int position) {
+        switch (position) {
+            case 0:
+                AllTabFragment allTabFragment = new AllTabFragment();
```
ViewPager holds 3 pages in memory by default. It means that if there are more that 3 tabs and user navigates to page #3 and then returns to page #1 the ViewPager will call this method again for the 1st index. It leads to recreation of the first fragment. And if there is any call to DB or to a server it will be re-invoked. Therefore, there is a reason to avoid such recalling, because it might be time or Internet data consuming. Could you please create more smart behavior here that will avoid such recalling?

PavelKapustsin commented on app/src/main/res/layout/all_tasks_tab.xml:
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
why did you use RelativeLayout here? what is this for?

PavelKapustsin replied:
why did you use RelativeLayout here? what is this for?

PavelKapustsin commented on app/src/main/res/layout/favorite_tasks_tab.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
why did you use RelativeLayout here? what is this for?

PavelKapustsin replied:
why did you use RelativeLayout here? what is this for?

PavelKapustsin commented on app/src/main/res/layout/main_activity.xml:
```diff
@@ -0,0 +1,45 @@
+    <android.support.v4.view.ViewPager
+        android:id="@+id/pager"
+        android:layout_width="match_parent"
+        android:layout_height="697dp"/>
```
Why did you add strict value of the element height?

PavelKapustsin commented on app/src/main/java/com/example/vanahel/tasksmanagerapplication/adapter/ViewPagerAdapter.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Override
+    public Fragment getItem(int position) {
```
ViewPager holds 3 pages in memory by default. It means that if there are more that 3 tabs and user navigates to page 3 and then returns to page 1 the ViewPager will call this method again for the 1st index. It leads to recreation of the first fragment. And if there is any call to DB or to a server it will be re-invoked. Therefore, there is a reason to avoid such recalling, because it might be time or Internet data consuming. Could you please create more smart behavior here that will avoid such recalling?

PavelKapustsin commented on app/src/main/res/layout/tool_bar.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.v7.widget.Toolbar
```
do you use this layout?

PavelKapustsin reviewed (changes requested):
I'ver written a few comments to you, could you please have a look?

LizavetaVanahel committed:
1. Tabs changed to Fragments

LizavetaVanahel committed:
Initial commit

LizavetaVanahel committed:
MVP

PavelKapustsin commented on app/build.gradle:
```diff
@@ -1,4 +1,5 @@
 apply plugin: 'com.android.application'
+apply plugin: 'com.jakewharton.butterknife'
```
Why do you use this plugin? It's needed if you use butteknife in a library only.

PavelKapustsin commented on app/build.gradle:
```diff
@@ -23,13 +24,15 @@ dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
         exclude group: 'com.android.support', module: 'support-annotations'
+        compile 'com.jakewharton:butterknife:8.7.0'
```
Are you sure about nesting butteknife library into androidTestCompile for espresso?

PavelKapustsin commented on app/src/main/java/com/example/vanahel/tasksmanagerapplication/NewTaskActivity.java:
```diff
@@ -1,73 +1,66 @@
+            task = getTask();
+            setTask(task);
+        }catch (Exception e){
             System.out.println("The fields are empty");
```
Don't use System.out in Android applications, use Log instead

PavelKapustsin commented on app/src/main/java/com/example/vanahel/tasksmanagerapplication/adapter/TasksRecyclerViewAdapter.java:
```diff
@@ -0,0 +1,60 @@
+        public TasksViewHolder (View view) {
+            super(view);
+
+            cardView = (CardView) view.findViewById(R.id.card_view);
```
Why didn't you use ButterKnife here?

PavelKapustsin commented on app/src/main/java/com/example/vanahel/tasksmanagerapplication/tab/AllTabFragment.java:
```diff
@@ -0,0 +1,71 @@
+
+    private AsyncTaskLoaderCallbacks asyncTaskLoaderCallbacks;
+    private TasksArrayAdapter allTasksArrayAdapter;
+    private static AllTabFragment allTabFragment;
```
Don't use static links on Fragments, Activities and other Android stuff that might have UI impact and\or have link on Context. Because it might lead to Memory Leaks.

PavelKapustsin commented on app/src/main/java/com/example/vanahel/tasksmanagerapplication/tab/FavoriteTabFragment.java:
```diff
@@ -0,0 +1,69 @@
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.favorite_tasks_tab, container, false);
```
Use ButterKnife everywhere

LizavetaVanahel committed:
1)sideNavigationMenu moving to the separate class
2)updating startActivityForResult in NewTaskActivity
3)ListItemMenuPresenter deleted and methods handling moved to TaskArrayAdapterPresenter
4)Calling Activity and Fragment methods through interface in TaskArrayAdapterPresenter

LizavetaVanahel committed:
1) Deleted static variables from TabFragments
2)Created TabFragment providers

LizavetaVanahel committed:
1) Added Internet storage
2)Push notification
3)Share option

LizavetaVanahel committed:
1) Changed logic of taking and sending screenshot
2)Added exceptions handing and exception message displaying

LizavetaVanahel committed:
External Storage permissions bug fixes

LizavetaVanahel committed:
External Storage permissions bug fixes
